File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        
        
        q = []
        q.append(root)
        
        ans = None
        
        while len(q) >0:
            
            node = q.pop(0)
            ans = node.val
            
            if node.right is not None:
                q.append(node.right)
            
            if node.left is not None:
                q.append(node.left)

        
        return ans
        

File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int findBottomLeftValue(TreeNode root) {
        
        
        int ans = 0;
        
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        q.add(root);
        
        while(!q.isEmpty()){
            
            TreeNode node = q.poll();
            
            ans = node.val;
            
            if(node.right !=null){
                q.add(node.right);
            }
            if(node.left !=null){
                q.add(node.left);
            }

        }
        
        return ans;
        
    }
}